### PR TITLE
Fix DAG export bug

### PR DESCRIPTION
**Pull request by ezraporter**
# Description
This PR fixes a bug causing users to receive an unclear error message when trying to export data access group information when they don't have the correct privileges.

# Proposed Changes 
- Wrap `redcap_dag_read()` in `try_redcapr()` to gracefully handle failures
- Add block to `try_redcapr()` to specifically handle the insufficient privileges error message
- Add related test cases
  -  This uses a new REDCap project with the API key called `REDCAPTIDIER_DAG_ACCESS_API`: https://redcap.chop.edu/redcap_v15.4.0/ProjectSetup/index.php?pid=75351

**Screenshots**

Screenshot 2025-05-30 at 9 57 04 AM

### Issue Addressed
**Closes #[provide issue number to link here]**

### PR Checklist
Before submitting this PR, please check and verify below that the submission meets the below criteria:

- [x] New/revised functions have associated tests
- [x] New/revised functions that update downstream outputs have associated static testing files (`.RDS`) updated under `inst/testdata/create_test_data.R`
- [x] New/revised functions use appropriate naming conventions
- [x] New/revised functions don't repeat code
- [x] Code changes are less than **250** lines total
- [x] Issues linked to the PR using [GitHub's list of keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] The appropriate reviewer is assigned to the PR
- [x] The appropriate developers are assigned to the PR
- [x] Pre-release package version incremented using `usethis::use_version()`

# Code Review
This section to be used by the reviewer and developers during Code Review after PR submission

### Code Review Checklist

- [ ] I checked that new files follow naming conventions and are in the right place
- [ ] I checked that documentation is complete, clear, and without typos
- [ ] I added/edited comments to explain "why" not "how"
- [ ] I checked that all new variable and function names follow naming conventions
- [ ] I checked that new tests have been written for key business logic and/or bugs that this PR fixes
- [ ] I checked that new tests address important edge cases

### CODE REVIEW

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -31,6 +31,7 @@ jobs:
       REDCAPTIDIER_LONGITUDINAL_DAG_API: ${{ secrets.REDCAPTIDIER_LONGITUDINAL_DAG_API }}
       REDCAPTIDIER_MIXED_STRUCTURE_API: ${{ secrets.REDCAPTIDIER_MIXED_STRUCTURE_API }}
       REDCAPTIDIER_MDC_API: ${{ secrets.REDCAPTIDIER_MDC_API }}
+      REDCAPTIDIER_DAG_ACCESS_API: ${{ secrets.REDCAPTIDIER_DAG_ACCESS_API }}
```
Comment (rsh52) [inline .github/workflows/R-CMD-check.yaml:34]: Annoying to add a whole other database and secret for this but c'est la vie.

File: R/utils.R [hunk 2]
```diff
@@ -805,19 +805,22 @@ try_redcapr <- function(expr, call = caller_env()) {
...
       )
       condition$class <- c("cannot_post", condition$class)
+    } else if (!is.null(out$outcome_message) && str_detect(
+      out$outcome_message,
+      "You must have 'API Export' privileges and 'Data Access Groups' privileges"
+    )
+    ) {
+      condition$info <- c(
+        "!" = "You do not have sufficient privileges to export data access groups.",
+        "i" = "Set `export_data_access_groups = FALSE` if you do not intend to export data access groups."
```
Comment (rsh52) [inline R/utils.R:815]: Perfect!

**Review by rsh52 — APPROVED** [review]
Thanks for adding this, all looks good to me. When you go to submit, just be sure to update `NEWS.md` to reflect these two bug fixes.
